janus/src/time.rs

```diff
@@ -1,22 +1,28 @@
 //! Utilities for timestamps and durations.
 
 use crate::message::Time;
-use chrono::{naive::NaiveDateTime, Utc};
+use chrono::Utc;
 use std::fmt::{Debug, Formatter};
 
 /// A clock knows what time it currently is.
-pub trait Clock: 'static + Clone + Copy + Debug + Sync + Send {
+pub trait Clock: 'static + Clone + Debug + Sync + Send {
```

Comment:
I don't think so: note that the lifetime reference in a trait bound doesn't affect the allowed lifetimes of references to objects implementing this trait, it bounds the lifetime of _internal_ references that the trait implementor might hold. So this is effectively saying "any references held by an implementor of the `Clock` trait must have lifetime at least as large as `'static`". This is needed because we have a number of functions that take a `Clock` implementation, move it into some structure that is itself `'static`, and return the structure -- if `Clock` wasn't `'static`, the returned structure might live longer than whatever reference is held by the `Clock` implementation.

See e.g. https://stackoverflow.com/questions/48504103/what-does-the-static-lifetime-mean-in-a-trait-bound-in-a-rust-future.

Reply:
Thanks, that's a very helpful explanation!